Added Korean README (#11)
klaytn/caver-js-examples: README.ko.md

# caver-js-examples

> **caver-js-examples** 은 더 쉽고 빠르게 Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))를 사용할 수 있도록 예제 소스코드를 제공하는 프로젝트입니다.

Reviewer: ```suggestion
**caver-js-examples**는 Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))를 더 쉽고 빠르게 사용할 수 있도록 예제 코드를 제공하는 프로젝트입니다.
```

다양한 상황에 맞는 예제코드를 확인하고 직접 실행해 볼 수 있습니다.

> ## 1. 선행되어야 하는 작업

Reviewer: ```suggestion
## 1. 선행 작업
```

> ### 1.1 실습을 위해 필요한 소프트웨어 설치

Reviewer: ```suggestion
### 1.1 소프트웨어 설치
```

> 본 프로젝트를 구성할 때 필요한 소프트웨어들을 기술합니다. 원활한 실습 진행을 위해 서는 [문서](https://ko.docs.klaytn.com/bapp/sdk/caver-js/getting-started#prerequisites)에 기술되어 있는 소프트웨어들은 실습 전에 설치해야 합니다.

Reviewer: ```suggestion
실습을 원활히 진행하려면 [문서](https://ko.docs.klaytn.com/bapp/sdk/caver-js/getting-started#prerequisites)에 작성된 소프트웨어를 먼저 설치해야 합니다. 해당 문서에는 본 프로젝트를 구성할 때 필요한 소프트웨어를 기술하고 있습니다.
```

### 1.2 KAS 회원가입
Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))는 Klaytn 네트워크를 구성하는 Node를 연결하여 사용할 수 있으며, caver-js-examples 프로젝트에서는 사용자들이 [EN](https://docs.klaytn.com/node/endpoint-node)을 직접 운영하지 않고 예제코드들을 테스트 할 수 있도록 [KAS](https://klaytnapi.com)의 [NODE API](https://refs.klaytnapi.com/en/node/latest)를 사용합니다.

> KAS를 사용하기 위해서는 [KAS](https://klaytnapi.com)에 가입해야 합니다. 가입하면 예제코드 실행에 필요한 `access key id`와 `secret access key`를 발급받을 수 있습니다.

Reviewer: ```suggestion
KAS를 사용하려면 먼저 [KAS](https://klaytnapi.com)에 가입하고, 예제코드 실행에 필요한 `AccessKey ID`와 `Secret AccessKey`를 발급 받습니다.
```

> 발급받은 `access key id`와 `secret access key`를 사용하는 방법은 [2. 사용법](https://github.com/klaytn/caver-js-examples/blob/master/README.ko.md#2-%EC%82%AC%EC%9A%A9%EB%B2%95)에 설명되어 있습니다.

Reviewer: ```suggestion
발급 받은 `AccessKey ID`와 `Secret AccessKey`의 사용 방법은 [2. 사용법](https://github.com/klaytn/caver-js-examples/blob/master/README.ko.md#2-%EC%82%AC%EC%9A%A9%EB%B2%95)에서 확인할 수 있습니다.
```

### 1.3 계정 준비
예제코드 실행을 위해서는 충분한 테스트넷 KLAY를 보유하고 있는 테스트 계정이 준비되어야 합니다. 아래는 테스트용 계정을 생성하고 5 테스트넷 KLAY를 충전하는 방법을 설명합니다.
예제코드에서는 Klaytn에 트랜잭션을 전송하기 때문에 최소 1개의 테스트 계정이 필요하며, 예제에 따라서 여러 개의 테스트 계정이 필요한 경우 아래의 절차대로 여러 개의 테스트계정을 생성할 수 있습니다. 또한 계정키를 업데이트해서 더이상 기존에 사용하던 테스트계정을 사용할 수 없는 경우에도 새롭게 테스트계정을 생성할 수 있습니다.
* https://baobab.wallet.klaytn.com/create 에서 테스트용 계정을 생성합니다. 
    * 생성된 계정의 주소와 개인키는 추후 테스트에 사용되기 때문에 따로 적어놔야 합니다.
    * 생성 시 저장한 키스토어 파일도 따로 보관해야 합니다. 일부 예제 시나리오에서는 키스토어 파일이 사용됩니다.
* [Baobab Klaytn Wallet의 Faucet](https://baobab.wallet.klaytn.com/faucet)에서 5 테스트넷 KLAY를 받습니다.

## 2. 사용법

> 1. 이 프로젝트를 클론합니다. `$ git clone https://github.com/klaytn/caver-js-examples.git`

Reviewer: ```suggestion
1. 이 프로젝트를 클론합니다. 터미널에서 `$ git clone https://github.com/klaytn/caver-js-examples.git`을 실행하세요.
```

> 2. 예제를 실행하기 위해서는 [KAS Console](https://console.klaytnapi.com/ko/security/credential)에서 발급받은 credential 과 위에서 발급받은 클레이튼 계정을 설정해야 합니다. 실습에 사용하는 계정은 테스트 용도의 계정만 사용하는 것을 권장합니다.

Reviewer: ```suggestion
2. 예제를 실행하려면 [KAS Console](https://console.klaytnapi.com/ko/security/credential)에서 발급받은 Credential(AccessKey ID와 Secret AccessKey)과 [Klaytn Wallet 테스트넷](https://baobab.wallet.klaytn.com/create)에서 발급받은 Klaytn 계정을 설정해야 합니다. 실습에 사용하는 계정은 테스트 용도로만 사용할 것을 권장합니다.
```

    * 옵션 1(권장): `caver-js-examples/.env` 파일의 내용에 정의합니다. 이 파일에 정의된 내용은 모든 시나리오에 공통 적용됩니다. 각 시나리오의 `index.js` 파일의 `loadEnv` 함수에서 `.env` 파일에 정의된 변수들을 읽어오는 코드가 있으니 참고하시기 바랍니다.
    * 옵션 2: 원하는 시나리오에 해당하는 `index.js` 파일을 열고 상단에 let 키워드로 선언된 변수들의 값을 채워줍니다.
3. 터미널에서 `node <Common Architecture 레이어>/<시나리오>/index.js`로 예제 소스코드를 실행합니다.
    * 예시 1: `$ node account/update_account_with_account_key_public/index.js`
    * 예시 2: `$ node contract/fee_delegation/index.js`

## 3. 프로젝트 구조
caver-js-examples 프로젝트의 구조에 대해 설명합니다.
### 3.1 CA-Layer/시나리오

> 디렉토리 트리의 가장 첫 번째 레벨에는 [Common Architecture를 구성하는 레이어들](https://kips.klaytn.com/KIPs/kip-34#layer-diagram-of-the-common-architecture)이 위치합니다. 아래 설명부터는 이를 `CA-Layers`라고 표기합니다.

Reviewer: ```suggestion
디렉토리 트리의 첫 번째 레벨에는 [Common Architecture를 구성하는 레이어](https://kips.klaytn.com/KIPs/kip-34#layer-diagram-of-the-common-architecture)가 위치합니다. 이하 `CA-Layers`라고 표기합니다.
```

# caver-js-examples

> **caver-js-examples** 프로젝트는 SDK를 사용하는 분들이 좀 더 쉽고 빠르게 SDK를 사용하실 수 있게끔 도움을 드리기 위한 프로젝트입니다.
> 다양한 상황에 맞는 예제코드를 직접 확인하시고 실행해보실 수 있습니다.

Reviewer 2: ```suggestion
**caver-js-examples** 은 더 쉽고 빠르게 Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))를 사용할 수 있도록 예제 소스코드를 제공하는 프로젝트입니다.
다양한 상황에 맞는 예제코드를 확인하고 직접 실행해 볼 수 있습니다.
```

## 1. 선행되어야 하는 작업
### 1.1 실습을 위해 필요한 소프트웨어 설치

> > 본 프로젝트를 구성할 때 사용한 소프트웨어들을 기술합니다. 원활한 실습 진행을 위해 아래 기술한 소프트웨어들은 실습 전에 설치 해주시기 바랍니다.
> * [Node.js](https://nodejs.org/en/download/) (아래 버전 중 하나를 권장드립니다.)
>     * lts/erbium ([12.21.0](https://nodejs.org/dist/latest-v12.x/))
>     * lts/fermium ([14.16.0](https://nodejs.org/dist/latest-v14.x/))
> * [npm](https://www.npmjs.com/get-npm)
> * [gcc-c++](https://gcc.gnu.org/)

Reviewer 2: ```suggestion
본 프로젝트를 구성할 때 필요한 소프트웨어들을 기술합니다. 원활한 실습 진행을 위해 서는 [문서](https://ko.docs.klaytn.com/bapp/sdk/caver-js/getting-started#prerequisites)에 기술되어 있는 소프트웨어들은 실습 전에 설치해야 합니다.
```
If not essential, how about using docs link to reduce duplicate contents for easier maintenance?

Author: @jimni1222 
Good idea :)

### 1.2 KAS 회원가입

> Caver SDK는 [Klaytn EndpointNode(이하 Klaytn EN)](https://docs.klaytn.com/node/endpoint-node)와 함께 사용할 수 있도록 만들어져 있습니다.
> [EN](https://docs.klaytn.com/node/endpoint-node)을 직접 운영하지 않아도 본 예제코드들을 테스트해볼 수 있도록, caver-js-examples 프로젝트는 [KAS](https://klaytnapi.com)의 [NODE API](https://refs.klaytnapi.com/en/node/latest)를 사용합니다.

Reviewer 2: ```suggestion
Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))는 Klaytn 네트워크를 구성하는 Node를 연결하여 사용할 수 있으며, caver-js-examples 프로젝트에서는 사용자들이 [EN](https://docs.klaytn.com/node/endpoint-node)을 직접 운영하지 않고 예제코드들을 테스트 할 수 있도록 [KAS](https://klaytnapi.com)의 [NODE API](https://refs.klaytnapi.com/en/node/latest)를 사용합니다.
```

> KAS를 사용하기 위해서는 [KAS](https://klaytnapi.com)에 가입하셔야 합니다. 가입하시면 예제코드 실행에 필요한 `access key id`와 `secret access key`를 발급받으실 수 있습니다.
> 발급받은 `access key id`와 `secret access key`는 [2. 사용법](https://github.com/klaytn/caver-js-examples/blob/master/README.ko.md#2-%EC%82%AC%EC%9A%A9%EB%B2%95) 섹션을 참고하셔서 적절한 위치에 값을 넣어주신 후 사용하시면 됩니다.

Reviewer 2: ```suggestion
KAS를 사용하기 위해서는 [KAS](https://klaytnapi.com)에 가입해야 합니다. 가입하면 예제코드 실행에 필요한 `access key id`와 `secret access key`를 발급받을 수 있습니다.
발급받은 `access key id`와 `secret access key`를 사용하는 방법은 [2. 사용법](https://github.com/klaytn/caver-js-examples/blob/master/README.ko.md#2-%EC%82%AC%EC%9A%A9%EB%B2%95)에 설명되어 있습니다.
```

### 1.3 계정 준비

> 5 KLAY 이상을 보유하고 있는 테스트 용도의 계정을 5개 이상 준비해주시기 바랍니다.

Reviewer 2: Can you explain why user has to create at least 5 accounts?

Author: Actually I think three would be enough.
The reason I said 5 is because of the use case below.
1. Fee Payer Account
2. Deployer Account (It doesn't matter if the Sender and Deployer accounts are the same.)
3. Sender Account
4. Sender Account to execute [account/update_account_with_account_key_weighted_multisig](https://github.com/klaytn/caver-js-examples/blob/master/account/update_account_with_account_key_weighted_multisig/boilerplate.js) scenario. When a user updates an account, that account is difficult to use for another scenarios.
5. Sender Account to execute [account/update_account_with_account_key_role_based](https://github.com/klaytn/caver-js-examples/blob/master/account/update_account_with_account_key_role_based/boilerplate.js) scenario. When a user updates an account, that account is difficult to use for another scenarios.

It's also based on my experience. When I ran the examples, I felt that 5 was enough.

> 5 KLAY 이상을 보유하고 있는 테스트 용도의 계정을 5개 이상 준비해주시기 바랍니다.

Reviewer 2: ```suggestion
예제코드 실행을 위해서는 충분한 테스트넷 KLAY를 보유하고 있는 테스트 계정이 준비되어야 합니다. 아래는 테스트용 계정을 생성하고 5 테스트넷 KLAY를 충전하는 방법을 설명합니다.
예제코드에서는 Klaytn에 트랜잭션을 전송하기 때문에 최소 1개의 테스트 계정이 필요하며, 예제에 따라서 여러 개의 테스트 계정이 필요한 경우 아래의 절차대로 여러 개의 테스트계정을 생성할 수 있습니다. 또한 계정키를 업데이트해서 더이상 기존에 사용하던 테스트계정을 사용할 수 없는 경우에도 새롭게 테스트계정을 생성할 수 있습니다.
```
How about this then?

> * https://baobab.wallet.klaytn.com/create 에서 테스트용 계정을 생성하실 수 있습니다.
> * 생성하신 테스트 계정으로 https://baobab.wallet.klaytn.com/faucet 에서 테스트용 5 KLAY를 충전하실 수 있습니다.
> * 향후 다시 테스트하실 때 생성하신 테스트용 계정들을 재사용하실 수 있으니 계정의 주소와 개인키를 기억할 수 있는 곳에 잘 보관해두시는 것을 추천드립니다.
> * 더불어 키스토어 파일도 가급적 보관해두시길 권장드립니다. 일부 예제 시나리오에서 사용이 됩니다.

Reviewer 2: ```suggestion
* https://baobab.wallet.klaytn.com/create 에서 테스트용 계정을 생성합니다. 
    * 생성된 계정의 주소와 개인키는 추후 테스트에 사용되기 때문에 따로 적어놔야 합니다.
    * 생성 시 저장한 키스토어 파일도 따로 보관해야 합니다. 일부 예제 시나리오에서는 키스토어 파일이 사용됩니다.
* [Baobab Klaytn Wallet의 Faucet](https://baobab.wallet.klaytn.com/faucet)에서 5 테스트넷 KLAY를 받습니다.
```

## 2. 사용법
1. 이 프로젝트를 클론합니다. `$ git clone https://github.com/klaytn/caver-js-examples.git`

> 2. [KAS](https://www.klaytnapi.com/ko/landing/main)에서 본인의 계정으로 발급받은 credential 정보와 실습에 사용하실 클레이튼 계정들의 주소와 개인키 값을 설정합니다. 실습에 사용하시는 계정은 테스트 용도의 계정 사용을 권장드립니다.

Reviewer 2: ```suggestion
2. 예제를 실행하기 위해서는 [KAS Console](https://console.klaytnapi.com/ko/security/credential)에서 발급받은 credential 과 위에서 발급받은 클레이튼 계정을 설정해야 합니다. 실습에 사용하는 계정은 테스트 용도의 계정만 사용하는 것을 권장합니다.
```

>     * 옵션 1(권장): `caver-js-examples/.env` 파일의 내용을 채워주도록 합니다. 이 파일에 채우신 내용은 모든 시나리오에 공통 적용됩니다. 각 시나리오의 `boilerplate.js` 파일의 `loadEnv` 함수에 `.env` 파일에 정의된 변수들을 읽어들이는 코드가 있으니 참고해주세요.
>     * 옵션 2: 원하는 시나리오에 해당하는 `boilerplate.js` 파일을 열고 상단에 let 키워드로 선언된 변수들의 값을 채워줍니다.

Reviewer 2: ```suggestion
    * 옵션 1(권장): `caver-js-examples/.env` 파일의 내용에 정의합니다. 이 파일에 정의된 내용은 모든 시나리오에 공통 적용됩니다. 각 시나리오의 `boilerplate.js` 파일의 `loadEnv` 함수에서 `.env` 파일에 정의된 변수들을 읽어오는 코드가 있으니 참고하시기 바랍니다.
    * 옵션 2: 원하는 시나리오에 해당하는 `boilerplate.js` 파일을 열고 상단에 let 키워드로 선언된 변수들의 값을 채워줍니다.
```

> 3. 사용하시는 터미널에서 원하는 시나리오를 정하신 뒤 `node <Common Architecture 레이어>/<시나리오>.js`와 같이 실행시켜줍니다.

Reviewer 2: ```suggestion
3. 터미널에서 `node <Common Architecture 레이어>/<시나리오>/boilerplate.js`로 예제 소스코드를 실행합니다.
```

    * 예시 1: `$ node account/update_account_with_account_key_public/boilerplate.js`
    * 예시 2: `$ node contract/fee_delegation/boilerplate.js`

## 3. 프로젝트 구조

> > caver-js-examples 프로젝트의 구조에 대한 이해를 돕기 위해 작성한 세션입니다.

Reviewer 2: ```suggestion
caver-js-examples 프로젝트의 구조에 대해 설명합니다.
```

### 3.1 CA-Layer/시나리오

> 디렉토리 트리의 가장 첫 번째 레벨에는 [Common Architecture를 구성하는 레이어들](https://kips.klaytn.com/KIPs/kip-34#layer-diagram-of-the-common-architecture)이 위치해 있습니다. 편의상 아래 설명부터는 `CA-Layers`라고 표기하도록 하겠습니다.

Reviewer 2: ```suggestion
디렉토리 트리의 가장 첫 번째 레벨에는 [Common Architecture를 구성하는 레이어들](https://kips.klaytn.com/KIPs/kip-34#layer-diagram-of-the-common-architecture)이 위치합니다. 아래 설명부터는 이를 `CA-Layers`라고 표기합니다.
```

```
.
├── abi
├── account
├── ...
└── wallet
```

> `CA-Layers` 바로 밑에는 **예제 시나리오들**이 위치하게 됩니다. 일례로 `account` 레이어에는 다음과 같은 예제 시나리오들이 위치해 있습니다.

Reviewer 2: ```suggestion
`CA-Layers` 바로 밑에는 **예제 시나리오들**이 위치하게 됩니다. 일례로 `account` 레이어 디렉토리 내부에는 다음의 예제 시나리오들이 위치합니다.
```

```
account
├── update_account_with_account_key_public
├── update_account_with_account_key_role_based
└── update_account_with_account_key_weighted_multisig
```
* 각 시나리오들은 단일 `boilerplate.js` 파일 그리고 해당 시나리오에 특정 파일이 필요한 경우 `resources` 디렉토리를 가지고 있습니다.

> * 이는 사용자 분들이 바로 시나리오 디렉토리에서 예제를 바로 실행해보실 수 있도록 구성하기 위함입니다.

Reviewer 2: ```suggestion
* 이는 사용자가 예제를 바로 실행해 볼 수 있도록 구성하기 위함입니다.
```

### 3.2 Summary

> 정리해보면 이 프로젝트는 여러 개의 `CA-Layer/시나리오`로 되어 있으며 사용자들이 원하는 시나리오를 프로젝트의 루트 디렉토리에서 `node CA-Layer/시나리오/boilerplate.js` 또는 해당 시나리오 디렉토리에서 `./node boilerplate.js`로 바로 예제코드를 실행해볼 수 있게 구성되어 있습니다.

Reviewer 2: ```suggestion
caver-js-example 프로젝트는 여러 개의 `CA-Layer/시나리오`로 되어 있으며, 프로젝트의 루트 디렉토리에서 `node CA-Layer/시나리오/boilerplate.js` 또는 해당 시나리오 디렉토리에서 `node ./boilerplate.js`로 예제코드를 실행할 수 있게 구성되어 있습니다.
```

## License

> **caver-java-examples** is released under the [MIT license](./LICENSE).

Reviewer 2: ```suggestion
**caver-js-examples** is released under the [MIT license](./LICENSE).
```

```
MIT License

> Copyright (c) 2021 caver-java-examples Authors

Reviewer 2: ```suggestion
Copyright (c) 2021 caver-js-examples Authors
```

# caver-js-examples
**caver-js-examples** 은 더 쉽고 빠르게 Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))를 사용할 수 있도록 예제 소스코드를 제공하는 프로젝트입니다.

> 다양한 상황에 맞는 예제코드를 확인하고 직접 실행해 볼 수 있습니다.

Reviewer: ```suggestion
다양한 상황에 맞는 예제 코드를 확인하고 직접 실행해 볼 수 있습니다.
```

## 1. 선행 작업
### 1.1 소프트웨어 설치
실습을 원활히 진행하려면 [문서](https://ko.docs.klaytn.com/bapp/sdk/caver-js/getting-started#prerequisites)에 작성된 소프트웨어를 먼저 설치해야 합니다. 해당 문서에는 본 프로젝트를 구성할 때 필요한 소프트웨어를 기술하고 있습니다.

### 1.2 KAS 회원가입

> Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))는 Klaytn 네트워크를 구성하는 Node를 연결하여 사용할 수 있으며, caver-js-examples 프로젝트에서는 사용자들이 [EN](https://docs.klaytn.com/node/endpoint-node)을 직접 운영하지 않고 예제코드들을 테스트 할 수 있도록 [KAS](https://klaytnapi.com)의 [NODE API](https://refs.klaytnapi.com/en/node/latest)를 사용합니다.

Reviewer: ```suggestion
Klaytn SDK([caver-js](https://github.com/klaytn/caver-js))는 Klaytn 네트워크를 구성하는 노드를 연결해 사용할 수 있는데, caver-js-examples 프로젝트에서는 [KAS](https://klaytnapi.com)의 [Node API](https://refs.klaytnapi.com/en/node/latest)를 사용합니다. 따라서 사용자는 [엔드포인트 노드(EN)](https://docs.klaytn.com/node/endpoint-node)를 직접 운영하지 않아도 예제코드를 테스트 할 수 있습니다.
```

KAS를 사용하기 위해서는 [KAS](https://klaytnapi.com)에 가입해야 합니다. 가입하면 예제코드 실행에 필요한 `access key id`와 `secret access key`를 발급받을 수 있습니다.
발급받은 `access key id`와 `secret access key`를 사용하는 방법은 [2. 사용법](https://github.com/klaytn/caver-js-examples/blob/master/README.ko.md#2-%EC%82%AC%EC%9A%A9%EB%B2%95)에 설명되어 있습니다.

### 1.3 계정 준비

> 예제코드 실행을 위해서는 충분한 테스트넷 KLAY를 보유하고 있는 테스트 계정이 준비되어야 합니다. 아래는 테스트용 계정을 생성하고 5 테스트넷 KLAY를 충전하는 방법을 설명합니다.

Reviewer: ```suggestion
예제 코드를 실행하려면 테스트넷 KLAY를 충분히 보유한 테스트 계정이 준비되어야 합니다. 
```

> 예제코드에서는 Klaytn에 트랜잭션을 전송하기 때문에 최소 1개의 테스트 계정이 필요하며, 예제에 따라서 여러 개의 테스트 계정이 필요한 경우 아래의 절차대로 여러 개의 테스트계정을 생성할 수 있습니다. 또한 계정키를 업데이트해서 더이상 기존에 사용하던 테스트계정을 사용할 수 없는 경우에도 새롭게 테스트계정을 생성할 수 있습니다.

Reviewer: ```suggestion
예제 코드 역시 Klaytn에 트랜잭션을 전송하기 때문에 최소 1개의 테스트 계정이 필요합니다. 만약 계정 키를 업데이트해서 기존에 사용하던 테스트 계정을 더이상 사용할 수 없다면 새로운 계정을 생성할 수 있습니다. 아래 설명에 따라 테스트 계정을 생성하고 5 테스트넷 KLAY를 충전하세요. 예제 종류에 따라 여러 개의 테스트 계정이 필요할 수 있으며, 이 경우에는 아래 절차를 반복해 여러 개의 테스트 계정을 생성할 수 있습니다.
```

> * https://baobab.wallet.klaytn.com/create 에서 테스트용 계정을 생성합니다. 

Reviewer: ```suggestion
1. [Klaytn Wallet 테스트넷](https://baobab.wallet.klaytn.com/create)에서 테스트 계정을 생성합니다. 
```

>     * 생성된 계정의 주소와 개인키는 추후 테스트에 사용되기 때문에 따로 적어놔야 합니다.

Reviewer: ```suggestion
    * 생성된 계정의 주소와 개인 키는 테스트에 활용되기 때문에 반드시 기록해두고 잊지 않도록 합니다.
```

>     * 생성 시 저장한 키스토어 파일도 따로 보관해야 합니다. 일부 예제 시나리오에서는 키스토어 파일이 사용됩니다.

Reviewer: ```suggestion
    * 생성 시 저장한 키스토어 파일을 별도 보관하세요. 일부 예제 시나리오에서 키스토어 파일을 활용합니다.
```

> * [Baobab Klaytn Wallet의 Faucet](https://baobab.wallet.klaytn.com/faucet)에서 5 테스트넷 KLAY를 받습니다.

Reviewer: ```suggestion
2. [Baobab Klaytn Wallet의 Faucet](https://baobab.wallet.klaytn.com/faucet)에서 5 테스트넷 KLAY를 충전합니다.
```

## 2. 사용법
1. 이 프로젝트를 클론합니다. `$ git clone https://github.com/klaytn/caver-js-examples.git`
2. 예제를 실행하기 위해서는 [KAS Console](https://console.klaytnapi.com/ko/security/credential)에서 발급받은 credential 과 위에서 발급받은 클레이튼 계정을 설정해야 합니다. 실습에 사용하는 계정은 테스트 용도의 계정만 사용하는 것을 권장합니다.

>     * 옵션 1(권장): `caver-js-examples/.env` 파일의 내용에 정의합니다. 이 파일에 정의된 내용은 모든 시나리오에 공통 적용됩니다. 각 시나리오의 `index.js` 파일의 `loadEnv` 함수에서 `.env` 파일에 정의된 변수들을 읽어오는 코드가 있으니 참고하시기 바랍니다.

Reviewer: @aeharvlee 파일의 내용에 Credential과 Klaytn 계정을 정의하는 건가요?

Author: @minishell 
네. `.env` 파일에는 KAS에 상요되는 Credential 정보와 실습에 필요한 Klaytn 계정들이 정의되어 있습니다.
사용할 Node API의 URL을 설정하는 부분도 있습니다 :)

Reviewer: ```suggestion
    * 옵션 1(권장): **caver-js-examples/.env** 파일에 발급 받은 Credential과 Klaytn 계정을 정의합니다. 이 파일에 정의된 내용은 모든 시나리오에 공통 적용됩니다. 각 시나리오의 **index.js** 파일 `loadEnv` 함수에서 **.env** 파일에 정의한 변수들을 읽어오는 코드가 있으니 참고하시기 바랍니다.
```

    * 옵션 2: 원하는 시나리오에 해당하는 `index.js` 파일을 열고 상단에 let 키워드로 선언된 변수들의 값을 채워줍니다.

> 3. 터미널에서 `node <Common Architecture 레이어>/<시나리오>/index.js`로 예제 소스코드를 실행합니다.

Reviewer: ```suggestion
3. 터미널에서 `node <Common Architecture 레이어>/<시나리오>/index.js`로 예제 코드를 실행합니다.
```

>     * 옵션 2: 원하는 시나리오에 해당하는 `index.js` 파일을 열고 상단에 let 키워드로 선언된 변수들의 값을 채워줍니다.

Reviewer: ```suggestion
    * 옵션 2: 원하는 시나리오에 해당하는 **index.js** 파일을 열고 상단에 let 키워드로 선언된 변수의 값을 채웁니다.
```

    * 예시 1: `$ node account/update_account_with_account_key_public/index.js`
    * 예시 2: `$ node contract/fee_delegation/index.js`
    
### 2.1 Hello World

> 시나리오에 관계 없이 바로 caver를 사용해보고 싶다면 `hello_world` 를 실행해보시기 바랍니다. 아래의 커맨드 중 하나를 사용해서 실행할 수 있습니다.

Reviewer: ```suggestion
시나리오에 관계없이 caver를 바로 사용해보고 싶다면 `hello_world`를 실행해보세요. 아래 명령어 중 하나를 사용해 실행할 수 있습니다.
```

* `$ cd hello_world && node ./index.js`
* `$ node hello_world/index.js`

> `caver-js-examples/hello_world/index.js` 파일의 `run` 함수에 caver를 사용하는 코드를 자유롭게 작성하고 테스트해보시기 바랍니다.

Reviewer: ```suggestion
`caver-js-examples/hello_world/index.js` 파일의 `run` 함수에 caver를 사용하는 코드를 자유롭게 작성하고 테스트해 보세요.
```

## 3. 프로젝트 구조

> caver-js-examples 프로젝트의 구조에 대해 설명합니다.

Reviewer: ```suggestion
caver-js-examples 프로젝트 구조를 설명합니다.
```

> `caver-js-examples/hello_world/index.js` 파일의 `run` 함수에 caver를 사용하는 코드를 자유롭게 작성하고 테스트해 보세요.

Reviewer: ```suggestion
**caver-js-examples/hello_world/index.js** 파일의 `run` 함수에 caver를 사용하는 코드를 자유롭게 작성하고 테스트해 보세요.
```

### 3.1 CA-Layer/시나리오

디렉토리 트리의 가장 첫 번째 레벨에는 [Common Architecture를 구성하는 레이어들](https://kips.klaytn.com/KIPs/kip-34#layer-diagram-of-the-common-architecture)이 위치합니다. 아래 설명부터는 이를 `CA-Layers`라고 표기합니다.
```
.
├── abi
├── account
├── ...
└── wallet
```

> `CA-Layers` 바로 밑에는 **예제 시나리오들**이 위치하게 됩니다. 일례로 `account` 레이어 디렉토리 내부에는 다음의 예제 시나리오들이 위치합니다.

Reviewer: ```suggestion
`CA-Layers` 하위 레벨에는 **예제 시나리오**가 위치하게 됩니다. 예를 들어 `account` 레이어 디렉토리 내부에 위치하는 예제 시나리오는 다음과 같습니다.
```

```
account
├── update_account_with_account_key_public
├── update_account_with_account_key_role_based
└── update_account_with_account_key_weighted_multisig
```

> * 각 시나리오들은 단일 `index.js` 파일 그리고 해당 시나리오에 특정 파일이 필요한 경우 `resources` 디렉토리를 가지고 있습니다.

Reviewer: ```suggestion
* 각 시나리오는 단일 `index.js` 파일을 포함하며, 해당 시나리오에 특정 파일이 필요한 경우 `resources` 디렉토리를 구성합니다.
```

> * 각 시나리오는 단일 `index.js` 파일을 포함하며, 해당 시나리오에 특정 파일이 필요한 경우 `resources` 디렉토리를 구성합니다.

Reviewer: ```suggestion
* 각 시나리오는 단일 **index.js** 파일을 포함하며, 해당 시나리오에 특정 파일이 필요한 경우 `resources` 디렉토리를 구성합니다.
```

* 이는 사용자가 예제를 바로 실행해 볼 수 있도록 구성하기 위함입니다.

### 3.2 Summary

> caver-js-example 프로젝트는 여러 개의 `CA-Layer/시나리오`로 되어 있으며, 프로젝트의 루트 디렉토리에서 `node CA-Layer/시나리오/index.js` 또는 해당 시나리오 디렉토리에서 `node ./index.js`로 예제코드를 실행할 수 있게 구성되어 있습니다.

Reviewer: ```suggestion
caver-js-example 프로젝트는 여러 개의 `CA-Layer/시나리오`로 이루어져 있으며, 프로젝트의 루트 디렉토리에서 `node CA-Layer/시나리오/index.js` 또는 해당하는 시나리오 디렉토리에서 `node ./index.js`로 예제 코드를 실행할 수 있도록 지원합니다.
```
